src/vanilla/utils/proxyMap.ts

```diff
@@ -130,24 +130,32 @@ export function proxyMap<K, V>(entries?: Iterable<[K, V]> | undefined | null) {
       indexMap.clear()
     },
     forEach(cb: (value: V, key: K, map: Map<K, V>) => void) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      this.epoch // touch property for tracking
       const map = getMapForThis(this)
       map.forEach((index, key) => {
         cb(this.data[index]!, key, this)
       })
     },
     *entries(): MapIterator<[K, V]> {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      this.epoch // touch property for tracking
       const map = getMapForThis(this)
       for (const [key, index] of map) {
         yield [key, this.data[index]!]
       }
     },
     *keys(): IterableIterator<K> {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      this.epoch // touch property for tracking
       const map = getMapForThis(this)
       for (const key of map.keys()) {
         yield key
       }
     },
     *values(): IterableIterator<V> {
+      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+      this.epoch // touch property for tracking
```

Comment:
I haven't looked at the test and others yet, but this feels wrong...
(If this fixes something, we may have misunderstood something why `this.index` didn't work. that was actually my concern back then.)

Reply:
This is (so far) what I think is going on:
1. Reactivity is triggered when a tracke)d property is accessed (correct me if I'm wrong here)
2. The epoch property changes whenever the proxyMap structure changes
3. By touching epoch in the iterator methods, we ensure they're re-evaluated when there are updates

Without having a tracked property being touched inside of the the methods, the items within any of the iterator methods were not updating when a new key is added or changed. E.G.

```typescript
const state = proxyMap()
const Component = () => {
  const snap = useSnapshot(state)

  return (
    <ul>
      <button onClick={() => addItemToMap()}>Add Item</button>
      {Array.from(snap.values()).map(item => (
        <li key={item.id}>{item.name}</li> {/* this was not updating on click */}
      ))}
    </ul>
  )
}
```

Reply:
Is this the same scenario as using `this.index` inside of the individual mutation methods? This is only calling the epoch property once per full iteration to make sure that changes to data are reflected. I could add in an `!isProxy` check to only access epoch property within those methods within a snapshot.

Reply:
Yeah, I think I get it. In the previous version (v2.0.0), it touches `this.data.length` behind the scenes, but with v2.1.x, we need to do it manually.